### PR TITLE
doc(readme) rework

### PR DESCRIPTION
**Pull request by piotch**
Update `readme` and `doc` to be clearer to (external) contributors.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,76 +6,122 @@
...
-Now `pytest tests/mysql` should run all the mysql tests properly.
+## Testing
+
+We are using `pytest` and various packages of its ecosystem, to install the testing dependencies run:
+
+```bash
+pip install -r requirements-testing.txt
+```
+
+As each connector is an independant plugin its test are written idenpendtly from the rest of the codebase. Run the tests for a specifc connector (`http_api` in this example) like this:
```
Comment (raphaelvignes) [inline README.md:77]:
```suggestion
As each connector is an independant plugin its test are written independently from the rest of the codebase. Run the tests for a specifc connector (`http_api` in this example) like this:
```

**Review by raphaelvignes — COMMENTED** [review]
Thx for the update !

File: README.md [hunk 2]
```diff
@@ -6,76 +6,122 @@
...
+Our CI does run all the tests for all the connectors, like this: 
 
-If you want to run the tests for all the connectors you can add all the dependencies by typing
-`pip install -e ".[all]"` and `make test`.
+```
+pip install -e ".[all]"
+make test
+```
+
+Some connectors are tested using mocks (cf. `trello`) other are tested by making calls to data providers (cf. `elasticsearch`) running on the system in docker containers. The required images are in the `tests/docker-compose.yml` file, they need to be pulled (cf. `pytest --pull`) to run the relevant tests.
```
Comment (hachichaud) [inline README.md]: > running on the system

Peut etre un peu vague ? de quel système on parle ?

File: README.md [hunk 3]
```diff
@@ -6,76 +6,122 @@
...
 `microstrategy`).
 
-If you have a container for your target system, please do not hesitate to add a docker image in
-the `docker-compose.yml`. You can then use the fixture `service_container` to automatically
-start the docker and shut it down for you!
+If you have a container for your target system, add a docker image in
+the `docker-compose.yml`, then use the `pytest` fixture `service_container` to automatically
+start the docker and shut it down for you when you are running tests.
+
+The fixture will not pull the image for you for each test runs, you need to pull the image on your machine  (at least once) using the `pytest --pull` option.
```
Comment (hachichaud) [inline README.md]: > The fixture will not pull

c'est pas plutot : "the test runner will not pull" ?

File: README.md [hunk 4]
```diff
@@ -112,30 +167,30 @@ CONNECTORS_CATALOGUE = {
...
+```
+
 ### Step 4 : Create a pull request
-Make sure your new code is properly formatted by typing `make lint`.
-If it's not, please use `make format`!
-You can now create a pull request!
+
+Make sure your new code is properly formatted by running `make lint`. If it's not, please use `make format`. You can now create a pull request.
 
 ## Publish
```
Comment (hachichaud) [inline README.md:193]: C'est pas réservé aux tokars ca ?
On pourrait peut etre le préciser

**Review by hachichaud — COMMENTED** [review]
Fixed a few typos directly.
+ added a few questions 

suggestions (can be done in other PRs) 

- add doc of `def describe(self, data_source: ToucanDataSource)` method needed for deep integration
- add doc for pydantic usage and explain form generation
- add contributing guidelines (PR, CI, quality requirements)

**Comment by hachichaud** [comment]
jai rebase pour les tests

**Review by raphaelvignes — APPROVED** [review]
Thx !

**Pull request merged**